### DISCUSSION TITLE
Drag and Drop Upload of Folders

### DISCUSSION THREAD

**Discussion by ghost** [inveniosoftware/product-rdm#147] [Ideas]
Package version (if known): <version>

## Describe the bug

When creating a new record, we try to upload a folder with subfolders using 'drag and drop'.
The folder and subfolder structure is not preserved.

## Steps to Reproduce
https://127.0.0.1/uploads/new
2. Drag and Drop a folder with subfolders
**Original folder-structure:**
![01-folder-structure](https://user-images.githubusercontent.com/74763324/110797730-b2f53480-8279-11eb-88a6-2d9480a285b2.png)

3.See error

**After drag and drop:**
![02-drag-and-drop](https://user-images.githubusercontent.com/74763324/110797743-b983ac00-8279-11eb-8396-94383b50bbc7.png)

**Saved and published record:**

![03-result](https://user-images.githubusercontent.com/74763324/110797828-d324f380-8279-11eb-9fc0-9b9c030453a7.png)

## Expected behavior

- It would be great, that folders and subfolders can be uploaded and not only files all on one level.
- Obliviously, you could upload a zip-instead, but then we do have the folder-structure as preview and cannot preview the files.
- If uploading folders is not possible (as it is currently the case with the 'Uploads files' Button / Window) the drag-and-drop should give me a warning or disable me from doing it completely and not just catching some files - we guess it just gets the files in the first sub directory.

[a comment by an automated account is not shown]

**Comment by lnielsen** [comment]
Moved to product board as it's a bigger feature that also involves deciding if we allow upload of folders
